File: maxSumSubArray.cpp
#include <iostream>
using namespace std;

int getMax(int a, int b)
{
    return (a > b ? a : b);
}

int getMaxSum(int arr[], int size)
{
    int maxValue = arr[0];
    int realMax = arr[0];
    for (int i = 0; i < size; i++)
    {
        maxValue = getMax(arr[i], maxValue + arr[i]);
        realMax = getMax(maxValue, realMax);
    }
    return realMax;
}

int main()
{
    int arr[1000], size;
    cout << "Enter the size of the array : ";
    cin >> size;
    for (int i = 0; i < size; i++)
    {
        cout << "Enter Element " << i + 1 << " : ";
        cin >> arr[i];
    }
    int maxValue = getMaxSum(arr, size);
    cout << "The max sum of the given array is : " << maxValue;
    return 0;
}

File: Max Sum Sub-Array.py
#Write a C++ program to find the maximum sum of a subarray in a given integer array. You are given an array of integers, and your task is to find the contiguous subarray (i.e., a subarray without skipping elements) with the largest sum.

n=int(input("Enter the size of array : "))
arr=[]
for i in range(n):
    arr.append(int(input("Enter the elements of array : ")))
for i in range(n):
    k=i
    j=i+1
    for j in range(n):
        b=arr[i]+arr[j]
        if(b<arr[j]):
            b=[]
            break
        else:
            b=arr[i]+arr[j]
        k+=2
    print(b)
    





#if(max(arr)>sum(arr)):
  #  print(max(arr))
#else:
   # print(sum(arr))


            